server/src/main/java/com/cloud/network/guru/ControlNetworkGuru.java

```diff
@@ -166,18 +167,25 @@ public boolean release(NicProfile nic, VirtualMachineProfile vm, String reservat
         assert nic.getTrafficType() == TrafficType.Control;
         HypervisorType hType = vm.getHypervisorType();
         if ( ( (hType == HypervisorType.VMware) || (hType == HypervisorType.Hyperv) )&& isRouterVm(vm)) {
+            // for now place this in the vmware specific part, but it miught be more generic and move up two or three lines
+            if (!VirtualNetworkApplianceManager.RemoveControlIpOnStop.valueIn(vm.getVirtualMachine().getDataCenterId())) {
+                if (s_logger.isDebugEnabled()) {
+                    s_logger.debug(String.format("not releasing %s from %s with reservationId %s.", nic, vm, reservationId));
```

Comment:
add the reason in the debug message ?

Reply:
Do you mean
```suggestion
                    s_logger.debug(String.format("not releasing %s from %s with reservationId %s, as we can´t get the DateCenter for it.", nic, vm, reservationId));
```
?

Reply:
as systemvm.release.control.ip.on.stop is set to false for the data center ?